lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -730,7 +747,9 @@ def token_details_from_authorization(authorization)
 
       def payment_details_from(payment_method)
         payment_details = {}
-        if payment_method.respond_to?(:number)
+        if payment_method.is_a?(NetworkTokenizationCreditCard)
```

Comment:
Like in here, it's typical to check on `NetworkTokenizationCreditCard` and it's source field to determine the type of payment method. So, something like,

```ruby
payment_method.is_a?(NetworkTokenizationCreditCard) && payment_method.source == :network_token
```

Reply:
@shasum thank you for your comments, I fixed this part in the latest commit